### PR TITLE
[Prim][PIR] softsign decomp

### PR DESCRIPTION
**Pull request by cubehan3**
### PR Category
Operator Mechanism

### PR Types
Others

### Description
Pcard-66975
添加 softsign 前向及反向拆解

### CODE REVIEW

File: paddle/fluid/primitive/rule/vjp/details.h [hunk 1]
```diff
@@ -2251,6 +2251,31 @@ void swiglu_grad(const Tensor& x,
   set_output<T>(x_grad, dx);
 }
 
+template <typename T>
+void softsign_grad(const Tensor& x, const Tensor& out_grad, Tensor* x_grad) {
+  if (!x_grad) return;
```
Comment (cyber-pioneer) [inline paddle/fluid/primitive/rule/vjp/details.h]: no need

File: paddle/fluid/primitive/rule/vjp/details.h [hunk 2]
```diff
@@ -2251,6 +2251,31 @@ void swiglu_grad(const Tensor& x,
...
+
+  Tensor x_tmp = x;
+  Tensor out_grad_tmp = out_grad;
+
+  bool need_cast = is_half_dtype(x.dtype());
+  if (need_cast) {
```
Comment (cyber-pioneer) [inline paddle/fluid/primitive/rule/vjp/details.h]: no need

File: paddle/fluid/primitive/rule/vjp/details.h (hunk 1 above)
Comment (cubehan3) [reply paddle/fluid/primitive/rule/vjp/details.h]: fix in commit [3ea2ae2](https://github.com/PaddlePaddle/Paddle/pull/66450/commits/3ea2ae2e2294d3870de05f88e148709f8f5ba147)

File: paddle/fluid/primitive/rule/vjp/details.h (hunk 2 above)
Comment (cubehan3) [reply paddle/fluid/primitive/rule/vjp/details.h]: fix in commit [3ea2ae2](https://github.com/PaddlePaddle/Paddle/pull/66450/commits/3ea2ae2e2294d3870de05f88e148709f8f5ba147)

**Review by cyber-pioneer — APPROVED** [review]

**Pull request merged**